app/utils/redis.ts

```diff
@@ -24,10 +25,15 @@ export async function encodeData(value: number[]) {
   return value;
 }
 
-export async function getSession(id: string) {
+export async function getSession(sessionId: string): Promise<SocketData> {
   const redis = getClient();
-  const value = await redis.hget(sessionKey, id);
-  return value === "alive" || value === "dead" ? value : null;
+  const sessionState = await redis.hget(sessionKey, sessionId);
+  if (sessionState === "alive" || sessionState === "dead") {
+    return { sessionId, sessionState };
+  }
+  const newSessionId = (sessionId = crypto.randomBytes(8).toString("hex"));
```

Comment:
is it necessary to reassign `sessionId` here? 
```suggestion
  const newSessionId = crypto.randomBytes(8).toString("hex");
```

Reply:
not at all, just a left over bit of code from moving things around